File: investigations/append-metadata/node/hello.js
const { MongoClient } = require('mongodb');

async function main() {
    const client = new MongoClient('mongodb://127.0.0.1:27017');
    client.appendMetadata({ name: "N1", version: "V1" });
    client.appendMetadata({ name: "N2", version: "V2" });
    client.appendMetadata({ name: "N1", version: "V2" });
    await client.connect();
    const result = await client.db('admin').command({ ping: 1 });
    console.log(result);
    await client.close();
}

main().catch(console.error);

File: investigations/append-metadata/python/hello.py
from pymongo import MongoClient
from pymongo.driver_info import DriverInfo

client = MongoClient("mongodb://localhost:27017")
client.append_metadata(DriverInfo(name="N1", version="V1"))
client.append_metadata(DriverInfo(name="N2", version="V2"))
client.append_metadata(DriverInfo(name="N1", version="V2"))
result = client.admin.command("ping")

